clang/lib/CodeGen/CGSYCLRuntime.cpp

```diff
@@ -44,8 +44,8 @@ static bool isPFWI(const FunctionDecl &FD) {
   if (!MD)
     return false;
   static std::array<Util::DeclContextDesc, 3> Scopes = {
-      Util::DeclContextDesc{clang::Decl::Kind::Namespace, "cl"},
       Util::DeclContextDesc{clang::Decl::Kind::Namespace, "sycl"},
+      Util::DeclContextDesc{clang::Decl::Kind::Namespace, "_V1"},
```

Comment:
Do you we have SYCL 1.2.1 compatibility tests?
https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:headers-and-namespaces
We should have them in SYCL-CTS and make sure that DPC++ complies.

Reply:
Are we talking source-compatible or binary-compatible? The former is being done with namespace alias as part of this PR (I might need to add an explicit test for that though), while the latter shouldn't (can't?) be done.

Reply:
I think we are talking about source-compatibility. Yes, please, add SYCL-1.2.1 compatibility test to make sure that when <CL/sycl.hpp> is included, we can still use data types defined in `cl::sycl` namespace in device code.

Reply:
I think we would need to account for both scopes (here and in Sema) for compatibility right?

Reply:
We need to remove these name based checks. Now that we're modifying namespaces, this isn't future proof at all. We should instead modify `sycl_special` attribute to accept an argument and use that to do these checks where required instead.

Reply:
> I think we would need to account for both scopes (here and in Sema) for compatibility right?

I don't think we do, but I'd like to hear your reasoning why you think we might need that.

> We need to remove these name based checks. ... We should instead modify `sycl_special` attribute...

That is way above my comfort level with the FE - I guess I'd need someone from FE to make that part of the change.
Also, IMO, it should be done in a separate patch (either before or after this PR).

Reply:
>> I don't think we do, but I'd like to hear your reasoning why you think we might need that.

If the `cl/syc`l header is used as @bader mentioned above and users declare types using `cl::sycl::aspect`, etc,  wouldn't compiler checks like `isDeviceAspectType` fail  since it checks hard-coded scope which is now `sycl::_V1::aspect`?

>>That is way above my comfort level with the FE - I guess I'd need someone from FE to make that part of the change.
Also, IMO, it should be done in a separate patch (either before or after this PR).

 I can help with that. What is the timeline for this change like?

Reply:
Using `cl/sycl.h` doesn't change ABI, only introduce namespace alias (https://godbolt.org/z/x17Toq5vY).

> I can help with that. What is the timeline for this change like?

I don't think there are hard requirement other than the 2023 release, but this PR gets merge conflicts pretty easily and I really wouldn't want it to sit uncommitted for long.

Reply:
Ok Thanks for explaining. I guess this should work then, as long as we don't use older versions of cl/sycl.h